paymentsheet/src/androidTest/java/com/stripe/android/paymentsheet/utils/FlowControllerTestRunner.kt

```diff
@@ -94,9 +103,75 @@ private fun runFlowControllerTest(
 
     scenario.moveToState(Lifecycle.State.RESUMED)
 
-    val testContext = FlowControllerTestRunnerContext(scenario, flowController)
+    val testContext = FlowControllerTestRunnerContext(manualIdlingResource, scenario, flowController)
     block(testContext)
 
     assertThat(countDownLatch.await(5, TimeUnit.SECONDS)).isTrue()
     scenario.close()
 }
+
+private class TestFlowControllerWrapper(
+    private val manualIdlingResource: ManualIdlingResource,
+    private val flowController: PaymentSheet.FlowController
+) : PaymentSheet.FlowController {
+    override var shippingDetails: AddressDetails?
+        get() = flowController.shippingDetails
+        set(value) {
+            flowController.shippingDetails = value
+        }
+
+    override fun configureWithPaymentIntent(
+        paymentIntentClientSecret: String,
+        configuration: PaymentSheet.Configuration?,
+        callback: PaymentSheet.FlowController.ConfigCallback
+    ) {
+        manualIdlingResource.setBusy()
```

Comment:
Should we not do this, but do a `waitForPresent`, that only needs to override present, then we can wait for that automatically in our fill in card method (for example).

Reply:
Depends on if we want to import this logic to all our other tests that use `PaymentSheetPage`. Right now, only `FlowController` has an actual use for waiting for the configuration to complete.